astroquery/utils/tap/core.py

```diff
@@ -67,34 +91,54 @@ def __init__(self, url=None, host=None, server_context=None,
+            if tap_context is None:
+                tap_context = tap
             if protocol == "http":
-                self.__connHandler = TapConn(False,
-                                             host,
-                                             server_context,
-                                             tap_context,
-                                             port,
-                                             sslport)
+                tap = TapConn(ishttps=False,
+                              host=host,
+                              server_context=server_context,
+                              tap_context=tap_context,
+                              upload_context=upload_context,
+                              table_edit_context=table_edit_context,
+                              data_context=data_context,
+                              datalink_context=datalink_context,
+                              port=port,
+                              sslport=sslport)
+                self.__connHandler = tap
             else:
                 # https port -> sslPort
-                self.__connHandler = TapConn(True,
-                                             host,
-                                             server_context,
-                                             tap_context,
-                                             port,
-                                             port)
+                tap = TapConn(ishttps=True,
+                              host=host,
+                              server_context=server_context,
+                              tap_context=tap_context,
+                              upload_context=upload_context,
+                              table_edit_context=table_edit_context,
+                              data_context=data_context,
+                              datalink_context=datalink_context,
+                              port=port,
+                              sslport=port)
+                self.__connHandler = tap
         else:
-            self.__connHandler = TapConn(default_protocol_is_https,
-                                         host,
-                                         server_context,
-                                         tap_context,
-                                         port,
-                                         sslport)
+            tap = TapConn(ishttps=default_protocol_is_https,
+                          host=host,
+                          server_context=server_context,
+                          tap_context=tap_context,
+                          upload_context=upload_context,
+                          table_edit_context=table_edit_context,
+                          data_context=data_context,
+                          datalink_context=datalink_context,
+                          port=port,
+                          sslport=sslport)
+            self.__connHandler = tap
         # if connectionHandler is set, use it (useful for testing)

```

Comment:
You are right. If connHandler is set, no need to create connections.
About arguments consistency, again, you are right. I did not include these checks because the code would grow with multiple if's and, as you have said, this is a quite internal initialization.